### ISSUE TITLE
[Bug] Batch with only padding is returned after all streams are consummed.

### ISSUE THREAD

**Issue by Jonathan2021** [etienne87/pytorch-stream-dataloader#3]
Suppose we have a single stream = `[1, 2, 3]`, a batch size of `1` and `bin_size` (the sequence length inside batches) >= `3`, `padding_mode = "zeros" ` and `padding_value = [0] * bin_size.`

Initialize `dataloader` (an instance of `StreamDataLoader`) with these and iterate on it.

```
for batch in dataloader:
  print(batch)
```

Printed Result:
```
[ [ 1, 2, 3, ....]]
[[0,0,0,....]]
```

An extra batch is returned with only padding in it. The expected behavior would be that only the 1st batch is returned right ?

**Comment by etienne87** [comment]
Thanks for reporting the bug! i will write tests for these use cases and let you know if i manage to resolve

**Comment by etienne87** [comment]
ok i think this is done, see code here https://github.com/etienne87/pytorch-stream-dataloader/blob/more_tests/tests/test_stream_dataloader.py#L163. Basically i just drop the last batch if it contains only padding data. is that acceptable?

**Comment by etienne87** [comment]
see MR https://github.com/etienne87/pytorch-stream-dataloader/pull/4

**Comment by Jonathan2021** [comment]
Hi Etienne,
Sorry for the late reply and thanks for the changes.
That does help when the padding mode is "zeros" but not when it is "data". It's more of a hack than a solution.

**Comment by etienne87** [comment]
I see, what you care is that the number of batches is deterministic?

**Comment by etienne87** [comment]
Ok, i propose this: https://github.com/etienne87/pytorch-stream-dataloader/blob/fix_padding_issue/pytorch_stream_dataloader/stream_dataset.py#L19 

User's data is wrapped up in a class called StreamData which specifies the status of the data (padding or not)

And i added both versions of the test (data and zeros) https://github.com/etienne87/pytorch-stream-dataloader/blob/fix_padding_issue/tests/test_stream_dataloader.py#L177

**Comment by Jonathan2021** [comment]
I think it does the trick ;) 
Thanks a bunch.
